test/set.js

```diff
@@ -25,9 +25,10 @@ assert.equal(result.stderr, 'ls: no such file or directory: file_doesnt_exist\n'
 
 // set -e
 var result = shell.exec('node -e \"require(\'../global\'); set(\'-e\'); ls(\'file_doesnt_exist\'); echo(1234);\"');
-assert.equal(result.code, 1);
+var uncaughtErrorExitCode = process.version.indexOf('v0.10') === 0 ? 8 : 1;
+assert.equal(result.code, uncaughtErrorExitCode);
 assert.equal(result.stdout, '');
-assert.equal(result.stderr, 'ls: no such file or directory: file_doesnt_exist\n');
+assert(result.stderr.indexOf('Error: ls: no such file or directory: file_doesnt_exist') >= 0);
```

Comment:
Instead of directly printing the message to stderr, we're now throwing an Error, so we end up at node's uncaught error handling. That's why you end up with "Error:" in front of the message.